modules/node-shared/src/main/scala/org/tessellation/node/shared/infrastructure/consensus/ConsensusManager.scala

```diff
@@ -18,44 +18,51 @@ import org.tessellation.node.shared.infrastructure.consensus.trigger.{ConsensusT
 import org.tessellation.node.shared.infrastructure.metrics.Metrics
 import org.tessellation.schema.node.NodeState
 import org.tessellation.schema.node.NodeState._
+import org.tessellation.schema.peer.Peer
 import org.tessellation.schema.peer.Peer.toP2PContext
-import org.tessellation.schema.peer.{Peer, PeerId}
-import org.tessellation.security.hash.Hash
 import org.tessellation.security.signature.Signed
 
 import eu.timepit.refined.auto._
+import monocle.Lens
 import org.typelevel.log4cats.slf4j.Slf4jLogger
 import retry.RetryDetails
 import retry.RetryPolicies.{constantDelay, fullJitter, limitRetries}
 import retry.syntax.all._
 
-trait ConsensusManager[F[_], Key, Artifact, Context] {
+trait ConsensusManager[F[_], Key, Artifact, Context, Status, Outcome, Kind] {
   def registerForConsensus(observationKey: Key): F[Unit]
 
   def startFacilitatingAfterDownload(observationKey: Key, lastArtifact: Signed[Artifact], lastContext: Context): F[Unit]
 
-  def startFacilitatingAfterRollback(lastKey: Key, lastArtifact: Signed[Artifact], lastContext: Context): F[Unit]
+  def startFacilitatingAfterRollback(lastKey: Key, initialOutcome: Outcome): F[Unit]
```

Comment:
shouldn't be either `initialKey / initialOutcome` or `lastKey / lastOutcome`?
or just `key, outcome`?

Reply:
I think lastKey/lastOutcome makes sense - will just verify if that naming reflects the usage of these values.

Reply:
I would actually leave it as it is because these names `lastKey` and `initialOutcome` do convey well the meaning of these values:
- `lastKey` is the ordinal of the snapshot we performed rollback to (or key of first incremental during genesis creation), so the name works well
- `initialOutcome` is the constructed outcome that simulates/mocks the outcome of the presumed previous consensus (for genesis there is no previous outcome, but we still construct it). We can't really call it a previous outcome because the actual previous outcome either didn't exist or may have been different - the outcome is not persisted between restarts. So the name `initialOutcome` also works well.

Reply:
you said the outcome is of the "presumed previous consensus". so it's also the `last` one 🤔

Reply:
you said the outcome is of the "presumed previous consensus". so it's also the `last` one 🤔

Reply:
you said the outcome is of the "presumed previous consensus". so it's also the `last` one 🤔

Reply:
Although it's probably not the same as actual last outcome. The facilitators field is set to the initial (rollback, genesis) facilitator some of the fields are set to default values whereas they could actually be populated in the actual previous outcome. It's different for the key because snapshot ordinal is a singular thing e.i. ordinal of 100 is just that ordinal of 100 (so we can extract it from the snapshot), whereas outcome is a composite thing consisting of multiple things and because we don't persist the outcome between the restart it needs to be reconstructed here. Hence I understand why it was called `initial` rather than `last`- not my change by the way :). I can change it if you think it still makes more sense to call it `last`.